hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -93,9 +103,13 @@ public List<String> getTransformersNames() {
   @Override
   public Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties) {
     Dataset<Row> dataset = rowDataset;
+    Option<Schema> incomingSchemaOpt = sourceSchemaOpt;
     for (TransformerInfo transformerInfo : transformers) {
       Transformer transformer = transformerInfo.getTransformer();
       dataset = transformer.apply(jsc, sparkSession, dataset, transformerInfo.getProperties(properties));
+      if (enableSchemaValidation) {
+        incomingSchemaOpt = validateAndGetTransformedSchema(transformerInfo, dataset, incomingSchemaOpt, jsc, sparkSession, properties);
```

Comment:
Why are we setting the incomingSchemaOpt to the targetSchemaOpt returned by the method here?

Reply:
This is due to chained transformer. Transformed schema of 1st transformer is input schema for 2nd transformer. Added a comment for it.

Reply:
I was able to get around this by calling the `transformedSchema` before apply within the `apply` API.